shapely/ops.py

```diff
@@ -179,6 +179,63 @@ def triangulate(geom, tolerance=0.0, edges=False):
     gc = geom_factory(func(geom._geom, tolerance, int(edges)))
     return [g for g in gc.geoms]
 
+
+def voronoi_diagram(geom, envelope=None, tolerance=0.0, edges=False):
+    """
+    Constructs a Voronoi Diagram [1] from the given geometry.
+    Returns a list of geometries.
+
+    Parameters
+    ----------
+        geom: the input geometry whose vertices will be used to calculate
+            the final diagram.
+        env: clipping envelope for the returned diagram, automatically
+            determined if None. The diagram will be clipped to the larger
+            of this envelope or an envelope surrounding the sites.
+        tolerance: sets the snapping tolerance used to improve the robustness
+            of the computation. A tolerance of 0.0 specifies that no
+            snapping will take place.
+        edges: If False (default), return Polygon regions. Else, return only
+            edges e.g. LineStrings.
+
+    GEOS documentation can be found at [2]
+
+    Returns
+    -------
+        List of geometries representing the Voronoi regions.
```

Comment:
In numpydocs you can include the return type right after the `Returns` header without any indentation. 

```
Returns
---------
list 
    geometries representing the Voronoi regions
``` 

There's a good example here: https://numpydoc.readthedocs.io/en/latest/example.html